node/overseer/overseer-gen/proc-macro/src/impl_builder.rs

```diff
@@ -446,7 +489,9 @@ pub(crate) fn impl_builder(info: &OverseerInfo) -> proc_macro2::TokenStream {
 			/// Complete the construction and create the overseer type.
 			pub fn build(self)
 				-> ::std::result::Result<(#overseer_name<S, #( #baggage_generic_ty, )*>, #handle), #error_ty> {
-				let connector = #connector ::default();
+				let connector = #connector ::with_signal_capacity(
```

Comment:
```suggestion
				let connector = #connector ::with_capacity(
```

Reply:
It's actually only the signal channel here, added a commit that cleans up the impl details a bit.